drake/core/Core.h

```diff
@@ -0,0 +1,106 @@
+//  - implements the typecast method operator Eigen::Matrix<ScalarType,size_at_compile,1>()
+//  - implements std::ostream& operator<<(std::ostream& os, const Vector& x)
+// (The methods/defs below make it possible (and hopefully easy) to simply use Eigen::Matrix types to model this concept)
+
+  template<typename T>
+  struct VectorTraits;
+
+  template<typename _Scalar, int _Rows, int _Cols, int _Options, int _MaxRows, int _MaxCols>
+  struct VectorTraits<Eigen::Matrix<_Scalar, _Rows, _Cols, _Options, _MaxRows, _MaxCols> > {
+    enum {
+      RowsAtCompileTime = _Rows
+    };
+  };
+
+  // In order to use Eigen types to model the Drake::Vector concept, they must accept only a single template parameter (ScalarType)
+  // The following structures provide a means for doing it. (e.g. via Drake::VectorBuilder<Rows>::VecType )
+
+  template <int Rows>
+  struct VectorBuilder {
+    template <typename ScalarType> using VecType = Eigen::Matrix<ScalarType,Rows,1>;
+  };
+  template <typename ScalarType> using EmptyVector = Eigen::Matrix<ScalarType,0,1>;
+  template <typename ScalarType> using UnusedVector = Eigen::Matrix<ScalarType,0,1>;
+
+
+  template <typename ScalarType, template <typename> class Vector1, template <typename> class Vector2>
+  class CombinedVector {
+  public:
+    const static size_t vec1_rows = VectorTraits<Vector1<ScalarType> >::RowsAtCompileTime,
+                        vec2_rows = VectorTraits<Vector2<ScalarType> >::RowsAtCompileTime;
+    CombinedVector() {};  // allow use of default constructors for vec1 and vec2, also
+    CombinedVector(const Eigen::Matrix<ScalarType,vec1_rows + vec2_rows,1>& x) : vec1(x.topRows(vec1_rows)), vec2(x.bottomRows(vec2_rows)) {};
+    CombinedVector(const Eigen::Matrix<ScalarType,vec1_rows,1>& x1, const Eigen::Matrix<ScalarType,vec2_rows,1>& x2) : vec1(x1), vec2(x2) {};
+
+    CombinedVector& operator=(const Eigen::Matrix<ScalarType,vec1_rows + vec2_rows,1>& x) {
+      vec1 = x.topRows(vec1_rows);
+      vec2 = x.bottomRows(vec2_rows);
+      return *this;
+    }
+
+    Vector1<ScalarType> first(void) const { return vec1; }
+    Vector2<ScalarType> second(void) const { return vec2; }
+
+    operator Eigen::Matrix<ScalarType,vec1_rows + vec2_rows,1> () const {
+      Eigen::Matrix<ScalarType,vec1_rows + vec2_rows,1> x;
+      x << static_cast<Eigen::Matrix<ScalarType,vec1_rows,1> >(vec1), static_cast<Eigen::Matrix<ScalarType,vec2_rows,1> >(vec2);
+      return x;
+    }
+
+    friend std::ostream& operator<<(std::ostream& os, const CombinedVector& x)
+    {
+      os << x.vec1 << std::endl;
+      os << x.vec2 << std::endl;
+      return os;
+    }
+
+    std::size_t size() { return vec1.size()+vec2.size(); }
+
+  private:
+    Vector1<ScalarType> vec1;
+    Vector2<ScalarType> vec2;
+  };
+
+  template <template <typename> class Vector1, template <typename> class Vector2>
```

Comment:
Although we never use the scalar type parameters for Vector1 and Vector2, it might still be a good idea to give them a name in such cases for readability.